codec-http2/src/main/java/io/netty/handler/codec/http2/PriorityStreamByteDistributor.java

```diff
@@ -379,46 +383,33 @@ long unallocatedStreamableBytesForTree() {
     /**
      * A connection stream visitor that delegates to the user provided visitor.
      */
-    private class WriteVisitor implements Http2StreamVisitor {
-        Writer writer;
-        RuntimeException error;
-
-        void writeAllocatedBytes(Writer writer) {
-            try {
-                this.writer = writer;
-                try {
-                    connection.forEachActiveStream(this);
-                } catch (Http2Exception e) {
-                    // Should never happen since the visitor doesn't throw.
-                    throw new IllegalStateException(e);
-                }
-
-                // If an error was caught when calling back the visitor, throw it now.
-                if (error != null) {
-                    throw error;
-                }
-            } finally {
-                error = null;
-            }
+    private final class WriteVisitor implements Http2StreamVisitor {
+        private final Deque<Writer> writers = new ArrayDeque<Writer>(1);
+        private Writer writer;
+
+        void writeAllocatedBytes(Writer writer) throws Http2Exception {
+            this.writer = writer;
+            writers.push(writer);
```

Comment:
We could potentially do this in a cheaper way (just throw if re-entry is detected, only conditionally create / use the `writers` queue if re-entry condition is detected, etc...) however I figured I should just start simple and discuss here.